mmyolo/models/layers/yolo_bricks.py

```diff
@@ -329,3 +334,324 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
+            block_channels = mid_channels
+            final_conv_in_channels = 2 * in_channels
+            final_conv_out_channels = 2 * in_channels
+        elif mode == 'no_change_channel':
+            mid_channels = in_channels // 4
+            block_channels = mid_channels
+            final_conv_in_channels = in_channels
+            final_conv_out_channels = in_channels
+        else:
+            mid_channels = in_channels // 2
+            block_channels = mid_channels // 2
+            final_conv_in_channels = in_channels * 2
+            final_conv_out_channels = in_channels // 2
+
+        self.main_conv = ConvModule(
+            in_channels,
+            mid_channels,
+            1,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg)
+
+        self.short_conv = ConvModule(
+            in_channels,
+            mid_channels,
+            1,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg)
+
+        self.blocks = nn.ModuleList()
+        for _ in range(num_blocks):
+            if mode == 'reduce_channel_2x':
+                internal_block = ConvModule(
+                    mid_channels,
+                    block_channels,
+                    3,
+                    padding=1,
+                    conv_cfg=conv_cfg,
+                    norm_cfg=norm_cfg,
+                    act_cfg=act_cfg)
+            else:
+                internal_block = nn.Sequential(
+                    ConvModule(
+                        mid_channels,
+                        block_channels,
+                        3,
+                        padding=1,
+                        conv_cfg=conv_cfg,
+                        norm_cfg=norm_cfg,
+                        act_cfg=act_cfg),
+                    ConvModule(
+                        block_channels,
+                        block_channels,
+                        3,
+                        padding=1,
+                        conv_cfg=conv_cfg,
+                        norm_cfg=norm_cfg,
+                        act_cfg=act_cfg))
+            mid_channels = block_channels
+            self.blocks.append(internal_block)
+
+        self.final_conv = ConvModule(
+            final_conv_in_channels,
+            final_conv_out_channels,
+            1,
+            conv_cfg=conv_cfg,
+            norm_cfg=norm_cfg,
+            act_cfg=act_cfg)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward process
+         Args:
+             x (Tensor): The input tensor.
+         """
+        x_short = self.short_conv(x)
+        x_main = self.main_conv(x)
+        block_outs = []
+        x_block = x_main
+        for block in self.blocks:
+            x_block = block(x_block)
+            block_outs.append(x_block)
+        x_final = torch.cat((*block_outs[::-1], x_main, x_short), dim=1)
+        return self.final_conv(x_final)
+
+
+class MaxPoolBlock(BaseModule):
```

Comment:
May find a better name because this block has not only maxpool but also conv layers